### PR TITLE
UnionDAO: union-of-none and resolve(null) fixes

### PR DESCRIPTION
**Pull request by mdittmer**
- The union of no DAOs is the empty read-only `NullDAO`;
- Fix NullDAO bug (and add test): When primary and secondary both `resolve(null)`, the `UnionDAO` was not calling `resolve(null)`.

### CODE REVIEW

File: lib/confluence/set_ops.es6.js [hunk 1]
```diff
@@ -252,9 +252,12 @@ foam.CLASS({
...
   listeners: [
     function onFind(data, o) {
-      if (data.errored === true || data.found !== null) return;
+      // Return if already reject()ed or resolve()d promise.
+      if (data.errored === true || data.found) return;
+      // Resolve with value of "o" if null previously found, or o is non-null.
+      var shouldResolve = data.found === null || o !== null;
       data.found = o;
-      if (o !== null) data.resolve(data.found);
+      if (shouldResolve) data.resolve(data.found = o);
```
Comment (arobins) [inline lib/confluence/set_ops.es6.js]: Why the |= o| here? Did you mean to delete the previous line so that the assignment was only done when resolving the promise?

Comment (mdittmer) [inline lib/confluence/set_ops.es6.js]: That's not right. Will fix.
